mesonbuild/modules/cuda.py

```diff
@@ -335,9 +335,9 @@ def _nvcc_arch_flags(self, cuda_version, cuda_arch_list='Auto', detected=''):
             arch, codev = re.fullmatch(
                 '([0-9]+\\.[0-9])(?:\\(([0-9]+\\.[0-9])\\))?', arch).groups()
 
-            if version_compare(arch, '<' + cuda_lo_limit_gpu_architecture):
+            if cuda_lo_limit_gpu_architecture and version_compare(arch, '<' + cuda_lo_limit_gpu_architecture):
```

Comment:
AFAICT there is always a low version, but not always an upper cap?

Reply:
(It wasn't immediately obvious to me what the bugfix was. Maybe that means it should be hinted at in the commit message?)

Reply:
@eli-schwartz Yes, that's right, I added `cuda_lo_limit_gpu_architecture` mostly for symmetry with `cuda_hi_limit_gpu_architecture`.

Reply:
Hmm. I don't think the symmetry is necessary, and it's a pointless check.

Reply:
@eli-schwartz Very well, I edited the patch to remove that and rebased.

Reply:
Not a huge deal, but commit messages usually say "why" instead of just "what". So if I were to write a commit message for this, I'd probably say "there isn't always a maximum version to restrict, which should be a no-op but instead crashed" rather than "verify that the variable is set".